gtsfm/utils/metrics.py

```diff
@@ -0,0 +1,197 @@
+"""Utilities to compute and save evaluation metrics.
+
+Authors: Akshay Krishnan
+"""
+import json
+import numpy as np
+import os
+from dask.delayed import Delayed
+from typing import Dict, List, Optional, Tuple, Union
+
+from gtsam import (Point3, Rot3, Pose3, Unit3)
+
+import gtsfm.utils.geometry_comparisons as comp_utils
+from gtsfm.common.sfm_result import SfmResult
+
+# A StatsDict is a dict from string to optional floats or their lists.
+StatsDict = Dict[str, Union[Optional[float], List[Optional[float]]]]
+
+
+def get_errors_statistics(errors: List) -> StatsDict:
+    """Computes statistics (min, max, median) on the given list of errors
+
+    Args:
+        errors: List of errors for a metric.
+
+    Returns: 
+        A dict with keys min_error, max_error, median_error, 
+        and errors_list mapping to the respective stats.
+    """
+
+    metrics = {}
+    valid_errors = [error for error in errors if error is not None]
```

Comment:
is numpy without a list comprehension faster?
```python
is_valid_bool = ~np.np.isnan(errors.astype(np.float32)
valid_errors = np.array(errors)[is_valid_bool]
```

Reply:
Apparently, it is not, I just tried it out. List: 2.19e-05 np: 9.08e-05 (in seconds). Probably it is the two iterations that it needs if done the np way?

Reply:
cool, thanks for checking